core/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -327,6 +327,33 @@ public void pause() {
         this.state.updateAndGet(s -> s.isRunning() ? State.PAUSED : s);
     }
 
+    /**
+     * Indicates whether this tracking processor can construct a new {@code Thread} to process
+     * another segment of events.
+     *
+     * @return {@code true} if this tracking processor can construct a new {@code Thread}, {@code false} otherwise
+     */
+    public boolean hasAvailableThreads() {
+        return maxThreadCount > activeProcessorThreads();
+    }
+
+    /**
+     * Blacklist a segment identifier only if it is currently claimed by this instance.
+     * This causes the segment to be released and to no longer to be claimed until it's removed from the black list.
+     *
+     * @param segmentId the id of the segment to be blacklisted
+     */
+    public void blacklistSegment(int segmentId) {
+        if (activeSegments.containsKey(segmentId)) {
```

Comment:
Does a segment have to be active to be able to blacklist it? What if you want to move a segment to a specific node, then all others would need to blacklist it for a specific amount of time, right?

Reply:
I think we should rename this method to "releaseSegment". It makes it more obvious for users what the method does. The fact that it is blacklisted as a result of that, is an implementation detail.